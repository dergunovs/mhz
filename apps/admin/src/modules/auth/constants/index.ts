export const API_POST_LOGIN = 'auth/login';
export const API_GET_CHECK_AUTH = 'auth/check';

export const URL_LOGIN = '/';
