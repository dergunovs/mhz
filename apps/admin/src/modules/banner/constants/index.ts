export const URL_BANNER = '/banners';
export const URL_BANNER_CREATE = `${URL_BANNER}/create`;
export const URL_BANNER_EDIT = `${URL_BANNER}/edit`;
