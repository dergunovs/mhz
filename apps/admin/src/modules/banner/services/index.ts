import { ComputedRef, Ref } from 'vue';

import { api, useQuery, useMutation, IPageQuery, convertParams } from 'mhz-helpers';
import { API_BANNER, IBanner, IBaseReply, IFilterData } from 'mhz-contracts';

export function getBanners(query: Ref<IPageQuery | number>) {
  async function fn() {
    const params = convertParams(query);

    const { data } = await api.get<{ data: IBanner[]; total: number; filters: IFilterData }>(API_BANNER, { params });

    return data;
  }

  return useQuery({ queryKey: [API_BANNER, query], queryFn: fn });
}

export function getBanner(id?: ComputedRef<string | string[]>) {
  async function fn() {
    if (!id?.value) return null;

    const { data } = await api.get<{ data: IBanner }>(`${API_BANNER}/${id.value}`);

    return data.data;
  }

  return useQuery({ queryKey: [API_BANNER, id], queryFn: fn });
}

export function postBanner(options: object) {
  async function fn(formData: IBanner) {
    const { data } = await api.post<IBaseReply>(API_BANNER, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_BANNER], mutationFn: fn, ...options });
}

export function updateBanner(options: object) {
  async function fn(formData: IBanner) {
    const { data } = await api.patch<IBaseReply>(`${API_BANNER}/${formData._id}`, formData);

    return data;
  }

  return useMutation({ mutationKey: [API_BANNER], mutationFn: fn, ...options });
}

export function deleteBanner(options: object) {
  async function fn(id?: string) {
    if (!id) return null;

    const { data } = await api.delete<IBaseReply>(`${API_BANNER}/${id}`);

    return data;
  }

  return useMutation({ mutationKey: [API_BANNER], mutationFn: fn, ...options });
}
