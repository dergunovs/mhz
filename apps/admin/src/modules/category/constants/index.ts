export const URL_CATEGORY = '/categories';
export const URL_CATEGORY_CREATE = `${URL_CATEGORY}/create`;
export const URL_CATEGORY_EDIT = `${URL_CATEGORY}/edit`;

export const CATEGORY_FIELD_TYPE_OPTIONS = ['string', 'number', 'boolean'];
