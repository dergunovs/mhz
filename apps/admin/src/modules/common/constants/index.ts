export const URL_MAIN = '/main';
export const URL_ERROR = '/404';
