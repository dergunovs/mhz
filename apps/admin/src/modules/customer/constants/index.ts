export const URL_CUSTOMER = '/customers';

export const API_CUSTOMER = 'customer';
