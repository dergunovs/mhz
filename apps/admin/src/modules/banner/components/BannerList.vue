<template>
  <UiTable
    :headers="tableHeaders"
    :isLoading="!props.banners?.length"
    :modelValue="props.modelValue"
    @update:modelValue="(value: ISortOption) => emit('update:modelValue', value)"
    @reset="(value: string) => emit('reset', value)"
    data-test="banner-list-table"
  >
    <template v-if="props.banners?.length">
      <tr v-for="banner in props.banners" :key="banner._id" data-test="banner-list-row">
        <td data-grow>
          <RouterLink :to="`${URL_BANNER_EDIT}/${banner._id}`" data-test="banner-list-link">
            {{ banner.product.title }}
          </RouterLink>
        </td>
        <td data-no-wrap>{{ banner.isActive }}</td>
        <td data-no-wrap>
          {{ formatDate(banner.dateCreated) }}
        </td>
        <td data-no-wrap>
          {{ formatDate(banner.dateUpdated) }}
        </td>
      </tr>
    </template>
  </UiTable>
</template>

<script setup lang="ts">
import { IBanner } from 'mhz-contracts';
import { UiTable } from 'mhz-ui';
import { formatDate, ISortOption } from 'mhz-helpers';

import { URL_BANNER_EDIT } from '@/banner/constants';

interface IProps {
  banners?: IBanner[];
  modelValue?: ISortOption;
}

const props = defineProps<IProps>();
const emit = defineEmits(['update:modelValue', 'reset']);

const tableHeaders = [
  { value: 'product', title: 'Product' },
  { value: 'isActive', title: 'Is active' },
  { value: 'dateCreated', title: 'Created' },
  { value: 'dateUpdated', title: 'Updated' },
];
</script>
