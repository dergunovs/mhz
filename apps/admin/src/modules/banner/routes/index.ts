import { URL_BANNER, URL_BANNER_CREATE, URL_BANNER_EDIT } from '@/banner/constants';

export const bannerRoutes = [
  {
    path: URL_BANNER,
    name: 'BannerList',
    component: () => import('@/banner/pages/BannerListPage.vue'),
  },
  {
    path: URL_BANNER_CREATE,
    name: 'BannerCreate',
    component: () => import('@/banner/pages/BannerCreatePage.vue'),
  },
  {
    path: `${URL_BANNER_EDIT}/:banner`,
    name: 'BannerEdit',
    component: () => import('@/banner/pages/BannerEditPage.vue'),
  },
];
