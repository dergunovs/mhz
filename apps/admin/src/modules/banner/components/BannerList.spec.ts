import { DefineComponent } from 'vue';
import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';

import BannerList from './BannerList.vue';

import { wrapperFactory } from '@/common/test';
import { URL_BANNER_EDIT } from '@/banner/constants';
import { BANNERS } from '@/banner/fixtures';

const MODELVALUE = { value: '_id', isAsc: true };

let wrapper: VueWrapper;

const bannerListTable = '[data-test="banner-list-table"]';
const bannerListRow = '[data-test="banner-list-row"]';
const bannerListLink = '[data-test="banner-list-link"]';

beforeEach(() => {
  wrapper = wrapperFactory(BannerList, {
    props: {
      banners: BANNERS,
      modelValue: MODELVALUE,
    },
  });
});

enableAutoUnmount(afterEach);

describe('BannerList', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(BannerList)).toBeTruthy();
  });

  it('shows rows if banner props', async () => {
    expect(wrapper.findAll(bannerListRow).length).toEqual(BANNERS.length);

    await wrapper.setProps({ banners: [] });

    expect(wrapper.find(bannerListRow).exists()).toBe(false);
  });

  it('generated link to banner', async () => {
    const LINK = `${URL_BANNER_EDIT}/${BANNERS[0]._id}`;

    expect(wrapper.find(bannerListLink).attributes('to')).toEqual(LINK);
  });

  it('shows banner content in cells', async () => {
    expect(wrapper.find(bannerListLink).text()).toEqual(BANNERS[0].product.title);
  });

  it('emits sort option by table emit', async () => {
    wrapper.findComponent<DefineComponent>(bannerListTable).vm.$emit('update:modelValue', MODELVALUE);

    expect(wrapper.emitted()['update:modelValue']).toHaveLength(1);
    expect(wrapper.emitted()['update:modelValue'][0]).toEqual([MODELVALUE]);
  });

  it('emits reset by table reset', async () => {
    wrapper.findComponent<DefineComponent>(bannerListTable).vm.$emit('reset', MODELVALUE.value);

    expect(wrapper.emitted()['reset']).toHaveLength(1);
    expect(wrapper.emitted()['reset'][0]).toEqual([MODELVALUE.value]);
  });
});
