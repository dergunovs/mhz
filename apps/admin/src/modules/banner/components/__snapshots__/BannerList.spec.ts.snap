// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BannerList > matches snapshot 1`] = `
"<div>
  <table headers="[object Object],[object Object],[object Object],[object Object]" isloading="false" modelvalue="[object Object]" lang="en" data-test="banner-list-table">
    <tbody>
      <tr data-test="banner-list-row">
        <td data-grow=""><a to="/banners/edit/6517b90000009354ff5e5a2a" data-test="banner-list-link">Samsung 500 GB 980 PRO</a></td>
        <td data-no-wrap="">true</td>
        <td data-no-wrap="">Sep 30, 2023</td>
        <td data-no-wrap="">0</td>
      </tr>
      <tr data-test="banner-list-row">
        <td data-grow=""><a to="/banners/edit/6517b8b800009354ff5e5a16" data-test="banner-list-link">MSI MPG SEKIRA 500X</a></td>
        <td data-no-wrap="">true</td>
        <td data-no-wrap="">Sep 30, 2023</td>
        <td data-no-wrap="">Sep 30, 2023</td>
      </tr>
      <tr data-test="banner-list-row">
        <td data-grow=""><a to="/banners/edit/6517b77c00009354ff5e5997" data-test="banner-list-link">GIGABYTE M32U</a></td>
        <td data-no-wrap="">true</td>
        <td data-no-wrap="">Sep 30, 2023</td>
        <td data-no-wrap="">0</td>
      </tr>
    </tbody>
  </table>
</div>"
`;
